### DISCUSSION TITLE
[Enhancement] Visual studio code supported development environment

### DISCUSSION THREAD

**Discussion by levelfivecoder** [dotnet/maui#46] [General]
## Summary
We require a light weight development environment for mobile app development and should support in Linux.

**Comment by dhindrik** [comment]
Scott Hunter showed how he used Visual Studio Code in his session. And this is not an MAUI thing, it is a tooling thing so this issue will probably be somewhere else. Also, it is not only related to MAUI, but it is also about .NET for iOS/Android/macOS.

**Comment by GiampaoloGabba** [comment]
VS Code will be supported as stated in the readme of this repo :)
Moreover MAUI will get full support for the dotnet CLI, which is awesome

**Comment by jsuarezruiz** [comment]
Yes, MAUI will have Visual Studio Code and dotnet CLI support.

**Comment by GeraudFabien** [comment]
The question is wether or not it will work with codespace. And if it does could the emulator be : 
- On the codespace container
- On other container
- On my machine
- And support real device

**Comment by chhinsras** [comment]
I have been developing Blazor/ASP.NET project with Visual Studio. But I prefer code on VS code. The VS code blazor tooling is really lagging and immature. The intellisense, the errors, the highlight syntax are really painful. _**MAYBE YOU GUYS SHARE YOU THOUGHT?**_

Developer friendly is really really important! Good technology but suck development expereience, somehow, we are afraid of using.

The new challeger has about to come in Q2 next year, MAUI. I love MAUI. Sound interesting and of course, we are using .NET ecosystem. It's fully supported on Visual Studio. However, I do prefer to use VS code for reasons. Taking lesson learnt from Flutter VS code support. The main IDE is android studio, but many developers out there in love with VS code. Some important and useful features should be included as well - VS code is Microsoft product and so MAUI ? it should be a perfect complement to one another. Following featuers have to be included are:
- [ ] Intellisense
- [ ] Hot Reload
- [ ] Friendly Errors
- [ ] ETC.

**Comment by melakhvision** [comment]
When to will we have the full support of vs code with .Net MAUI ? Like react native, flutter and so. VS Code is more lightweight. 
By using making it fully usable with Visual Studio it restricted the audience like Linux user can't use it and we are forced to use visual studio even we don't lke it.

I don't understand one thing.

- VS code is made by Microsoft and .Net MAUI is created by Microsoft. What is the problem of making the two fully compatible when you own both ?
Google and facebook doesn't own VS code but each one makes his framework fully compatible to VS code.
.Net MAUI is a powerfull framewor as as .Net developer it make life easier but the development expereience sucks. 
I stopped using Xamarin many month before .Net MAUI was relase because of visual studio.
